### PR TITLE
Example and E2E test case for dynamic concurrency

### LINKED ISSUE #45: add e2e test cases for dynamic concurrency feature

### PR DESCRIPTION
**Pull request by SrinivasChilveri**
What this PR does / why we need it:
This PR adds dynamic concurrency feature e2e test case and sample example

added some validation in cmd/genectl/parser/parser.go

Which issue(s) this PR fixes (optional, in fixes #(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged):
Fixes #45

### CODE REVIEW

File: cmd/genectl/parser/parser.go [hunk 1]
```diff
@@ -71,6 +71,15 @@ func ValidateWorkflow(workflow *Workflow) ErrorList {
 
 		// validate depends
 		allErr = append(allErr, ValidateDepends(jobName, job.Depends, workflow.Jobs)...)
+
+		if (len(job.Commands) == 0) &&
+			((len(job.CommandsIter.Vars) == 0) && (len(job.CommandsIter.VarsIter) == 0)) {
+
+			err := fmt.Errorf("workflow.job.Commands or  (job.CommandsIter.Vars and job.CommandsIter.VarsIter)"+
```
Comment (hzxuzhonghu) [inline cmd/genectl/parser/parser.go]: `workflow.job.Commands` all lowercase

File: test/e2e/genectl.go [hunk 2]
```diff
@@ -91,6 +91,15 @@ var _ = DescribeGene("genectl", func(gtc *GeneTestContext) {
...
 		glog.Infof("output: %v", output)
 	})
+
+	It("sub workflow with get_result", func() {
+		createVolumeAndClaim("example/simple-sample-getresult/sample-pv.yaml", "example/simple-sample-getresult/sample-pvc.yaml", ns, kubeClient)
+
+		By("Execute sub workflow command")
+		cmd := NewGenectlCommand("sub", "workflow", "example/simple-sample-getresult/simple-sample-getresult.yaml", "--tool-repo="+ToolRepo)
+		output := cmd.ExecOrDie()
+		glog.Infof("output: %v", output)
```
Comment (hzxuzhonghu) [inline test/e2e/genectl.go]: you should verify the result, should include: the jobs and the output, and the job exection sequence.

File: cmd/genectl/parser/parser.go (hunk 1 above)
Comment (SrinivasChilveri) [reply cmd/genectl/parser/parser.go]: changed

File: test/e2e/genectl.go (hunk 2 above)
Comment (SrinivasChilveri) [reply test/e2e/genectl.go]: done

**Comment by SrinivasChilveri** [comment]
Old test cases related to the test/e2e/genectl.go file file are success but the executions were not sucess because when we are  using the genectl  the executions are created in default name space but the persistent volumes are created in the dynamic namespace so executions were not success. so changed the old & new test case also.

File: cmd/genectl/parser/parser.go [hunk 3]
```diff
@@ -71,6 +71,15 @@ func ValidateWorkflow(workflow *Workflow) ErrorList {
 
 		// validate depends
 		allErr = append(allErr, ValidateDepends(jobName, job.Depends, workflow.Jobs)...)
+
+		if (len(job.Commands) == 0) &&
+			((len(job.CommandsIter.Vars) == 0) && (len(job.CommandsIter.VarsIter) == 0)) {
+
+			err := fmt.Errorf("workflow.job.commands or  (job.commandsiter.vars and job.commandsiter.varsiter)"+
```
Comment (hzxuzhonghu) [inline cmd/genectl/parser/parser.go]: one of `xxxxxx` must be specified

File: example/simple-sample-getresult/README.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+## Overview
+
+This is a simple example to demonstrate how to submit a workflow using `genectl` with get_result
```
Comment (hzxuzhonghu) [inline example/simple-sample-getresult/README.md]: Add the link to [get_result]()

**Comment by hzxuzhonghu** [comment]
some nits, otherwise /lgtm

File: cmd/genectl/parser/parser.go (hunk 3 above)
Comment (SrinivasChilveri) [reply cmd/genectl/parser/parser.go]: k

File: example/simple-sample-getresult/README.md (hunk 4 above)
Comment (SrinivasChilveri) [reply example/simple-sample-getresult/README.md]: k

File: example/simple-sample-getresult/README.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+ * Ensure your tool repo has been set correctly.
+
+## Command
+
+```bash
+$ genectl sub workflow simple-sample-getresult.yaml
+```
+
+
+MoreInfo](https://kubegene.io/docs/design/dynamic-concurrency/dynamic-concurrency.md)
```
Comment (hzxuzhonghu) [inline example/simple-sample-getresult/README.md]:
```suggestion
[MoreInfo](https://kubegene.io/docs/design/dynamic-concurrency/dynamic-concurrency.md)
```

**Pull request merged**